consumer/spring-websocket-consumer/src/main/java/org/springframework/cloud/fn/consumer/websocket/trace/InMemoryTraceRepository.java

```diff
@@ -41,35 +43,50 @@ public class InMemoryTraceRepository {
 
 	private final List<Trace> traces = new LinkedList<>();
 
+	private final Lock tracesLock = new ReentrantLock();
+
 	/**
 	 * Flag to say that the repository lists traces in reverse order.
 	 * @param reverse flag value (default true)
 	 */
 	public void setReverse(boolean reverse) {
-		synchronized (this.traces) {
+		try {
+			this.tracesLock.lock();
```

Comment:
I don't think we need any locking in setters at all.

Reply:
Alright, then I’m removing the synchronized keyword as well. Is it correct?

Reply:
@artembilan I think locking is necessary here because the add method on line 86 uses the reverse and capacity values. add method requires locking, as any changes to reverse or capacity while the add method is being called could result in an inconsistent state. Therefore, I believe we should continue using locking.

Reply:
What you are explaining is correct, but that's not realistic. 
This class is created only once on Application context initialization.
Those setters are called at this point and only once.
Calling them at runtime is really mocking of the lifecycle of the instance of this class.

We should just agree that we are in Spring, so there are some undocumented rules for singleton beans where we just initialize them once. Plus the instance of this class is created by the configuration in this module.
You just need to drop the jar into your classpath - and everything will be auto-configured for you.

Why do we need the extra logic which may not have power in 99% of usage?